6/Api/src/Startup.cs

```diff
@@ -0,0 +1,36 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Api
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddHostedService<ClusterClientHostedService>();
+            services.AddTransient<IHostedService>(_ => _.GetService<ClusterClientHostedService>());
+            services.AddTransient(_ => _.GetService<ClusterClientHostedService>().Client);
```

Comment:
Ah no, I think this is going to be way too slow to instantiate the cluster client for every request. What is your current max RPS? Let's test it with a singleton client and then try and create a pool of those clients. I feel that the pool will be the fastest options, we just need to determine how many clients in the pool we will need.